spec/formatting.md

```diff
@@ -18,3 +18,57 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided to the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is invalid.
+- **Resolution errors** occur when the runtime value of a part of a message
+  cannot be determined.
+- **Unresolved variable errors** are a sub-category of resolution errors,
+  and occur when a variable reference cannot be resolved.
+- **Formatting errors** are thrown by expression handlers,
+  for example when encountering a value with an unsupported type
+  or an internally inconsistent set of options.
```

Comment:
1) Those two are subtypes of resolution errors. Should be nested under it.
2) What about selector errors? When none of the variants match
3) I'm wondering if the name "formatting" is appropriate. Would you consider broken message reference to be a formatting error?

Reply:
The first two points I'll need to address separately, but re "formatting", I would consider a broken message ref to be a resolution rather than formatting error.

Reply:
So, is it like unresolved variable?

Reply:
Pretty much, yes. To clarify that, should add explicit verbiage to indicate that an expression handler is able to emit both kinds of errors.